pageserver/src/tenant.rs

```diff
@@ -1704,7 +1704,10 @@ impl Tenant {
 
         let mut target_config_file = VirtualFile::open_with_options(
             target_config_path,
-            OpenOptions::new().write(true).create_new(first_save),
+            OpenOptions::new()
+                .truncate(true)
```

Comment:
I'm a bit concerned about durability here, its possible that after truncate pageserver can crash and we'll end up with empty config file. Postgres has [durable_rename](https://github.com/neondatabase/postgres/blob/39e3d745b3701a3f47f40412fbec62cbb01a42bf/src/common/file_utils.c#L346) function for that, we dont have common counterpart on the rust side, but rather code in in place where needed, for example [here](https://github.com/neondatabase/neon/blob/main/pageserver/src/storage_sync2/download.rs#L36)

IMO this can be done separately. Before the PR config endpoint was not really usable without some serious tweaking of the process. So lets do it separately

Also, just to note, I think this code was adapted from [save_metadata](https://github.com/neondatabase/neon/blob/9a6c0be823fb1b11f7d0ab6bb5a98605387c08d5/pageserver/src/tenant/metadata.rs#L232) here. And metadata save dont need durable rename because it is always the same size (512bytes) and we overwrite it each time assuming 512 bytes are written do disk atomically, because it should be one disk sector. And there is a check that we write indeed write 512 bytes. Tenant config is different, we dont have such checks in place. Prior to this PR strategy was to overwrite, and this is fine as long as tenant config is small, but id doesnt work because config size can change so we get this artifact with partial overwriting.

Reply:
so what you suggest is to 
1) create the config file in a config_tmp file
2) flush it
3) rename it to config 
if all succeeds return OK() ?

Reply:
@LizardWizzard  opened Issue https://github.com/neondatabase/neon/issues/3062 for tracking